extmod/machine_signal.c

```diff
@@ -114,6 +114,21 @@ STATIC mp_obj_t signal_make_new(const mp_obj_type_t *type, size_t n_args, size_t
     return MP_OBJ_FROM_PTR(o);
 }
 
+STATIC void signal_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    machine_signal_t *self = MP_OBJ_TO_PTR(self_in);
+    const mp_print_fun_t pin_print_function = (mp_print_fun_t)MP_OBJ_TYPE_GET_SLOT_OR_NULL(mp_obj_get_type(self->pin), print);
```

Comment:
This first bit can just be

```
mp_obj_print_helper(print, self->pin, PRINT_STR);
```

That will save 56 bytes, bringing this change to +80 bytes, which I think is not unreasonable.

Reply:
Thanks.
I hope I understood your suggestion correctly.

Reply:
Yep exactly that, thanks!